invenio_app/config.py

```diff
@@ -18,22 +18,89 @@
 websites.
 """
 
-RATELIMIT_DEFAULT = '5000/hour'
-"""Default rate limit.
+from __future__ import absolute_import, print_function, unicode_literals
 
-.. note:: Overwrite
-   Flask-Limiter <https://flask-limiter.readthedocs.io/en/stable/>`_
-   configuration.
+from invenio_app.limiter import set_rate_limit
+
+RATELIMIT_APPLICATION = set_rate_limit
+"""Global rate limit."""
+
+RATELIMIT_STRATEGY = 'moving-window'
+"""The rate limiting strategy to use.
+
+The strategy used here is the most consistant but also expensive one.
+If you are experiencing performance issues due to the increased Redis
+traffic, you can replace it with another one from the following
+`Flask-Limiter strategies
+<https://flask-limiter.readthedocs.io/en/stable/#ratelimit-strategy>`_.
 """
 
 RATELIMIT_HEADERS_ENABLED = True
-"""Enable rate limit headers. (Default: ``True``)
+"""Enable rate limit headers. (Default: ``True``)"""
 
-.. note:: Overwrite
-   Flask-Limiter <https://flask-limiter.readthedocs.io/en/stable/>`_
-   configuration.
+RATELIMIT_STORAGE_URL = 'memory://'
+"""Provide your Redis URL if you are rate limiting a multithreaded application.
```

Comment:
What do you think about having a generic description in this first line, then in the body what is actually in the note and then inside the note add the need to use Redis if you are rate-limiting a multithreaded application? For example:

```
"""Storage backend to store rate-limiting information.

Memory is used by default if no value is provided.
For more information regarding the mentioned above configuration values and
their available options, you can see the `Flask-Limiter configuration
<https://flask-limiter.readthedocs.io/en/stable/#configuration>`_.

.. note::

    You should provide a Redis URL if you are rate-limiting a multithreaded
    application.
```